src/Core/Models/Business/SecretsManagerSubscriptionUpdate.cs

```diff
@@ -1,56 +1,99 @@
     /// Usually this is what the organization is billed for
     /// </summary>
-    public int SmServiceAccountsExcludingBase { get; set; }
+    public int SmServiceAccountsExcludingBase { get; }
 
     /// <summary>
     /// The new autoscale limit for service accounts, expressed as a total (not an adjustment).
     /// This may or may not be the same as the current autoscale limit.
     /// </summary>
-    public int? MaxAutoscaleSmServiceAccounts { get; set; }
+    public int? MaxAutoscaleSmServiceAccounts { get; }
 
     public bool SmSeatsChanged => SmSeatsAdjustment != 0;
     public bool SmServiceAccountsChanged => SmServiceAccountsAdjustment != 0;
-    public bool MaxAutoscaleSmSeatsChanged { get; set; }
-    public bool MaxAutoscaleSmServiceAccountsChanged { get; set; }
+    public bool MaxAutoscaleSmSeatsChanged { get; }
+    public bool MaxAutoscaleSmServiceAccountsChanged { get; }
+
+    public SecretsManagerSubscriptionUpdate(
+        Organization organization,
+        int seatAdjustment, int? maxAutoscaleSeats,
+        int serviceAccountAdjustment, int? maxAutoscaleServiceAccounts)
+    {
+        var secretsManagerPlan = Utilities.StaticStore.GetSecretsManagerPlan(organization.PlanType);
+        if (secretsManagerPlan == null)
+        {
+            throw new NotFoundException("Invalid Secrets Manager plan.");
+        }
+
+        var newTotalSeats = organization.SmSeats.GetValueOrDefault() + seatAdjustment;
+        var autoscaleSeatsChanged = maxAutoscaleSeats.HasValue
+                                    && maxAutoscaleSeats != organization.MaxAutoscaleSmSeats.GetValueOrDefault();
+
+        var newTotalServiceAccounts = organization.SmServiceAccounts.GetValueOrDefault() + serviceAccountAdjustment;
+        var autoscaleServiceAccountsChanged = maxAutoscaleServiceAccounts.HasValue
+                                              && maxAutoscaleServiceAccounts != organization.MaxAutoscaleSmServiceAccounts.GetValueOrDefault();
+
+        OrganizationId = organization.Id;
+
+        SmSeats = newTotalSeats;
+        SmSeatsAdjustment = seatAdjustment;
+        SmSeatsExcludingBase = newTotalSeats - secretsManagerPlan.BaseSeats;
+        MaxAutoscaleSmSeats = maxAutoscaleSeats;
+
+        SmServiceAccounts = newTotalServiceAccounts;
+        SmServiceAccountsAdjustment = serviceAccountAdjustment;
+        SmServiceAccountsExcludingBase = newTotalServiceAccounts - secretsManagerPlan.BaseServiceAccount.GetValueOrDefault();
+        MaxAutoscaleSmServiceAccounts = maxAutoscaleServiceAccounts;
+
+        MaxAutoscaleSmSeatsChanged = autoscaleSeatsChanged;
+        MaxAutoscaleSmServiceAccountsChanged = autoscaleServiceAccountsChanged;
+    }
+
+    public SecretsManagerSubscriptionUpdate(Organization organization, int serviceAccountAdjustment)
+        : this(organization, seatAdjustment: 0, maxAutoscaleSeats: null, serviceAccountAdjustment, maxAutoscaleServiceAccounts: null)
```

Comment:
I think this is a poor candidate for a constructor overload, because the `int` argument is so general. When we add an interface for adjusting seats, it'll look the same:

```cs
    public SecretsManagerSubscriptionUpdate(Organization organization, int seatAdjustment)
```

...and that'll clash with this one.

So I would delete this overload and just use the main constructor above.

Reply:
Oh that makes sense, thanks!